### PR TITLE
Add Half support for range, logspace, logit, median, nanmedian, kthvalue, poisson, cummax, cummin, prod, cumprod, histc, logcumsumexp, vander, cross, aten2, logaddexp, logaddexp2, hypot, and nextafter on CPU

### PR DESCRIPTION
**Pull request by CaoE**
Add Half support for range, logspace, logit, median, nanmedian, kthvalue, poisson, cummax, cummin, prod, cumprod, histc, logcumsumexp, vander, cross, aten2, logaddexp, logaddexp2, hypot, and nextafter on CPU.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @Xia-Weiwen @ngimel

### CODE REVIEW

File: test/test_binary_ufuncs.py [hunk 1]
```diff
@@ -3804,12 +3805,19 @@ def _test_atan2_with_size(size, device):
...
+                    rtol = 0
+                    atol = 0.02
+                else:
+                    rtol = 0
+                    atol = 0.001
+                a_16 = a.to(dtype=lowp_dtype)
+                b_16 = b.to(dtype=lowp_dtype)
+                actual_16 = a_16.atan2(b_16)
+                self.assertEqual(actual_16, actual.to(dtype=lowp_dtype))
+                self.assertEqual(expected, actual_16.view(-1), exact_dtype=False, rtol=0, atol=0.02)
```
Comment (jgong5) [inline test/test_binary_ufuncs.py:3820]: you are not using the `rtol` and `atol` variables set previously?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: test/test_binary_ufuncs.py (hunk 1 above)
Comment (CaoE) [reply test/test_binary_ufuncs.py]: Thanks for your comments. Fixed it.

**Review by jgong5 — APPROVED** [review]

**Comment by CaoE** [comment]
@cpuhrsch Could you please review this PR ? Thank you.

**Comment by cpuhrsch** [comment]
@CaoE - I know this might be annoying, but can we split this up a bit more?

While the number of lines changed isn't high, the number of ops is. That'll make it harder to bisect if there's compounding issues.

**Comment by CaoE** [comment]
@cpuhrsch Thanks for your comments. I split this PR into https://github.com/pytorch/pytorch/pull/112124,  https://github.com/pytorch/pytorch/pull/112131, https://github.com/pytorch/pytorch/pull/112132,  https://github.com/pytorch/pytorch/pull/112135, and https://github.com/pytorch/pytorch/pull/112138

**Pull request closed without merging**